src/shared/gpt.h

```diff
@@ -10,6 +10,48 @@
 /* maximum length of gpt label */
 #define GPT_LABEL_MAX 36
 
+typedef enum PartitionDesignator {
+        PARTITION_ROOT, /* Primary architecture */
+        PARTITION_ROOT_SECONDARY, /* Secondary architecture */
+        PARTITION_ROOT_OTHER, /* Any architecture not covered by the primary or secondary architecture. */
+        PARTITION_USR,
+        PARTITION_USR_SECONDARY,
+        PARTITION_USR_OTHER,
+        PARTITION_HOME,
+        PARTITION_SRV,
+        PARTITION_ESP,
+        PARTITION_XBOOTLDR,
+        PARTITION_SWAP,
+        PARTITION_ROOT_VERITY, /* verity data for the PARTITION_ROOT partition */
+        PARTITION_ROOT_SECONDARY_VERITY, /* verity data for the PARTITION_ROOT_SECONDARY partition */
+        PARTITION_ROOT_OTHER_VERITY,
+        PARTITION_USR_VERITY,
+        PARTITION_USR_SECONDARY_VERITY,
+        PARTITION_USR_OTHER_VERITY,
+        PARTITION_ROOT_VERITY_SIG, /* PKCS#7 signature for root hash for the PARTITION_ROOT partition */
+        PARTITION_ROOT_SECONDARY_VERITY_SIG, /* ditto for the PARTITION_ROOT_SECONDARY partition */
+        PARTITION_ROOT_OTHER_VERITY_SIG,
+        PARTITION_USR_VERITY_SIG,
+        PARTITION_USR_SECONDARY_VERITY_SIG,
+        PARTITION_USR_OTHER_VERITY_SIG,
+        PARTITION_TMP,
+        PARTITION_VAR,
+        PARTITION_USER_HOME,
+        PARTITION_LINUX_GENERIC,
```

Comment:
why have this?

linux generic partitions do not have an inherent purpose, they only acquire one if they are is no partition with a proper purpose on a medium, and they are basically the only partition that could be considered a root fs.

Thus, I would argue this should not be a designator of its own, because we will never use it as such. We'll use it as PARTITION_ROOT only.

I think the GptPartitionType .designator field should just be set to _PARTITION_DESIGNATOR_INVALID for partitions of this type, because they really have no inherent designated purpose. And afaics your patch you never actually make use of PARTITION_LINUX_GENERIC except for setting it, hence nothing is lost.

(similar PARTITION_USER_HOME actually)